src/main/java/com/uber/jenkins/phabricator/coverage/CoberturaCoverageProvider.java

```diff
@@ -42,6 +50,37 @@ protected CodeCoverageMetrics getCoverageMetrics() {
         return convertCobertura(getCoverageResult());
     }
 
+    @Override
+    public Map<String, List<Integer>> readLineCoverage() {
+        String basePath = "";
+        FilePath workspace = getBuild().getWorkspace();
+        if (workspace != null) {
+            basePath = workspace.getRemote();
+        }
+        File[] reports = getCoberturaReports(getBuild());
+        CoberturaXMLParser parser = new CoberturaXMLParser(basePath);
+        return parseReports(parser, reports);
+    }
+
+    public Map<String, List<Integer>> parseReports(CoberturaXMLParser parser, File[] reports) {
+        if (reports == null) {
+            return null;
+        }
+        for (File coberturaXmlReport : reports) {
+            try {
+                // TODO support merging multiple coverage results
+                return parser.parse(coberturaXmlReport);
+            } catch (IOException e) {
+                e.printStackTrace();
+            } catch (ParserConfigurationException e) {
+                e.printStackTrace();
+            } catch (SAXException e) {
+                e.printStackTrace();
```

Comment:
@jjx is there a better way to handle all of these? I feel like catching the base `Exception` is wrong but this also feels convoluted.